plutus-core/plutus-ir/test/transform/inline/letApp

```diff
@@ -0,0 +1,27 @@
+(let
+    (nonrec)
+    (termbind (strict) (vardecl appNum (con integer)) (con integer 4))
+    (termbind
+      (strict)
+      (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
+      (builtin addInteger)
+    )
+    (termbind
+      (strict)
+      (vardecl addInteger (fun (con integer) (fun (con integer) (con integer))))
+      (lam x (con integer) (lam y (con integer) [ [ addInteger x ] y ]))
+    )
+    (termbind
+      (strict)
+      (vardecl funApp (fun (con integer) (con integer)))
+      (lam
+        eta
+        (con integer)
+        [
+          [ AddInteger appNum ]
+          eta
+        ]
+      )
+    )
+    [ funApp (con integer 5) ]
+)
```

Comment:
I actually think we should move away from crafting the tests by hand, because we are not expecting users to do that and there are other factors affecting the output PIR code. See also [here](https://input-output.atlassian.net/browse/PLT-1489).

Reply:
I said this in our meeting but I _strongly_ disagree with this. The point of these test cases is to test that the inliner is doing what we expect it to do. They absolutely _should_ be small and synthetic so that they're easy to inspect. They are _not_ supposed to be realistic examples, we have other tests for that.

Reply:
These are all generated by the compiler and they are small enough to read I think? Eg, I'd be more comfortable if we have tests including `Bool` instead of only `Integer` even though `Bool` is longer. In fact, to actually test non-single occurrences they'll need to be longer still......maybe need to stick with `Integer` for those. I'll see. It's true we want them tractable!

Reply:
I think we should try to test one thing at a time, as much as possible. If we don't think that having `Bool` in there will affect the behaviour of the inliner, then it's just noise that makes this test case harder to understand, IMO. 

I tend to think of test cases like this as witnessing a specific piece of behaviour, and so you want them to be as small and direct as possible while still showing the behaviour that you want.

Reply:
Right, but I'm suggesting that we want to be precise about what _statements_ we're verifying. I think it's more helpful to verify lots of small specific statements than fewer larger complex statements. Adding random additional complexity that we don't particularly expect to have an effect just muddles things.

If you want to have testcases that _specifically_ check whether using other datatypes has any effect, then sure, we can do that. But don't mix it up with the tests for other things, IMO.

Also obviously we can't test all combinations of "doing X other thing has no effect", we have to be judicious. I would only write such a test if I thought there was a reason such a thing _could_ interfere, but e.g. I'd written code to guard against it and I wanted to verify that it worked.